azure-pipelines.yml

```diff
@@ -1,342 +1,404 @@
-    - 'All'
-    - '00'
-    - '01'
+      - "All"
+      - "00"
+      - "01"
 
   - name: location
     displayName: Location
     type: string
-    default: 'UK South'
+    default: "UK South"
     values:
-    - 'UK South'
-    - 'UK West'
+      - "UK South"
 
-  - name: env
-    displayName: Environment
-    type: string
-    default: 'sbox'
-    values:
-    - ithc
-    - preview
-    - demo
-    - sbox
-    - perftest
-    - aat
-    - ptlsbox
-    - ptl
-    - prod
+  - name: environment_components
+    type: object
+    default:
+      - env: "sbox"
+        serviceConnection: "DCD-CFTAPPS-SBOX"
+        backendAzureRmResourceGroupName: "azure-control-sbox-rg"
+        aksKeyVault: "dcdcftappssboxkv"
+        terraformSubscriptionID: "b72ab7b7-723f-4b18-b6f6-03b0f2c6a1bb"
+      - env: "preview"
+        dependsOn: "sbox"
+        serviceConnection: "DCD-CFTAPPS-DEV"
+        backendAzureRmResourceGroupName: "azure-control-dev-rg"
+        aksKeyVault: "dcdcftappsdevkv"
+        terraformSubscriptionID: "8b6ea922-0862-443e-af15-6056e1c9b9a4"
+      - env: "demo"
+        dependsOn: "sbox"
+        serviceConnection: "DCD-CFTAPPS-DEMO"
+        backendAzureRmResourceGroupName: "azure-control-demo-rg"
+        aksKeyVault: "dcdcftappsdemokv"
+        terraformSubscriptionID: "d025fece-ce99-4df2-b7a9-b649d3ff2060"
+      - env: "ithc"
+        dependsOn: "sbox"
+        serviceConnection: "DCD-CFTAPPS-ITHC"
+        backendAzureRmResourceGroupName: "azure-control-ithc-rg"
+        aksKeyVault: "dcdcftappsithckv"
+        terraformSubscriptionID: "62864d44-5da9-4ae9-89e7-0cf33942fa09"
+      - env: "ptlsbox"
+        dependsOn: "sbox"
+        serviceConnection: "DTS-CFTSBOX-INTSVC"
+        backendAzureRmResourceGroupName: "azure-control-ptlsbox-rg"
+        aksKeyVault: "dtscftptlsbox"
+        terraformSubscriptionID: "1497c3d7-ab6d-4bb7-8a10-b51d03189ee3"
+      - env: "perftest"
+        dependsOn: "sbox"
+        serviceConnection: "DCD-CFTAPPS-TEST"
+        backendAzureRmResourceGroupName: "azure-control-test-rg"
+        aksKeyVault: "dcdcftappstestkv"
+        terraformSubscriptionID: "8a07fdcd-6abd-48b3-ad88-ff737a4b9e3c"
+      - env: "aat"
+        dependsOn: "sbox"
+        serviceConnection: "DCD-CFTAPPS-STG"
+        backendAzureRmResourceGroupName: "azure-control-stg-rg"
+        aksKeyVault: "dcdcftappsstgkv"
+        terraformSubscriptionID: "96c274ce-846d-4e48-89a7-d528432298a7"
+      - env: "ptl"
+        dependsOn: "ptlsbox"
+        serviceConnection: "DTS-CFTPTL-INTSVC"
+        backendAzureRmResourceGroupName: "azure-control-ptl-rg"
+        aksKeyVault: "dtscftptl"
+        terraformSubscriptionID: "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
+      - env: "prod"
+        dependsOn: "aat"
+        serviceConnection: "DCD-CFTAPPS-PROD"
+        backendAzureRmResourceGroupName: "azure-control-prod-rg"
+        aksKeyVault: "dcdcftappsprodkv"
+        terraformSubscriptionID: "8cbc6f36-7c56-4963-9d36-739db5d00b27"
```

Comment:
We agreed to only run prod if all nonprod and sbox passed I believe, is it possible to add multiple stages to this dependsOn?

Reply:
I know this is technically dependent on sbox && aat passing, but would be nice to rely on everything so we catch any possible errors before prod

Reply:
https://github.com/hmcts/aks-cft-deploy/pull/613/commits/f0cada7880bdac1850b5906f6977a5769982d4b6 is `nonprod` a thing?

Reply:
This https://github.com/hmcts/aks-cft-deploy/pull/613/commits/b875d85dbc67e6d86aaf5ec7e5a1da057282ce5c is the same as it is at the moment but explicit (as aat depends on sbox anyway)

I was more meaning along the lines of depends on aat, perftest, ithc, demo, ptl

Reply:
tried to do this but adding it on the stage. messing around creates a cycle in the dependency graph.